lib/runners/processor/clang_tidy.rb

```diff
@@ -0,0 +1,99 @@
+module Runners
+  class Processor::ClangTidy < Processor
+    Schema = StrongJSON.new do
+      let :runner_config, Schema::BaseConfig.base.update_fields { |fields|
+        fields.merge!({
+          apt: enum?(string, array(string)),
+          'include-path': enum?(string, array(string))
+        })
+      }
+
+      let :issue, object(
+        severity: string,
+      )
+    end
+
+    register_config_schema(name: :clang_tidy, schema: Schema.runner_config)
+
+    # @see https://github.com/github/linguist/blob/775b07d40c04ef6e0051a541886a8f1e30a892f4/lib/linguist/languages.yml#L532-L535
+    # @see https://github.com/github/linguist/blob/775b07d40c04ef6e0051a541886a8f1e30a892f4/lib/linguist/languages.yml#L568-L584
+    VALID_EXTENSIONS = [".c", ".cpp", ".c++", ".cc", ".cp", ".cxx"].freeze
+    GLOB_HEADERS = "**/*.{h,h++,hh,hpp,hxx,inc,inl,ipp,tcc,tpp}".freeze
+
+    def setup
+      deploy_packages
+      yield
+    end
+
+    def analyzer_bin
+      "clang-tidy"
+    end
+
+    def analyze(changes)
+      issues = []
+
+      changes
+        .changed_paths
+        .select { |path| VALID_EXTENSIONS.include?(path.extname.downcase) }
+        .map{ |path| relative_path(working_dir / path, from: current_dir) }
+        .reject { |path| path.to_s.start_with?("../") } # reject files outside the current_dir
+        .each do |path|
+          stdout, = capture3!(analyzer_bin, path.to_s, "--", *option_include_path,
```

Comment:
[suggestion] For example, how about linting multiple files together?

```suggestion
        .each_slice(100) do |paths| # Avoid too many arguments 
          stdout, = capture3!(analyzer_bin, *paths.to_map(&:to_path), "--", *option_include_path,
```

https://ruby-doc.org/core/Enumerable.html#each_slice-method

Reply:
I'm stuck. Results of `capture3!()` with multiple files are different from results of `clang-tidy` on `bash`.

(1) Three issues found on `bash`.
```shell
$ clang-tidy baz/test.cp foo/テスト.C++ test.CC -- -I bar/myinc
...
/home/kentaro/depot/runners/test/smokes/clang_tidy/option_root_dir/src/baz/test.cp:5:15: warning: Value stored to 't' during its initialization is never read [clang-analyzer-deadcode.DeadStores]
...
/home/kentaro/depot/runners/test/smokes/clang_tidy/option_root_dir/src/foo/テスト.C++:6:5: warning: 2nd function call argument is an uninitialized value [clang-analyzer-core.CallAndMessage]
...
/home/kentaro/depot/runners/test/smokes/clang_tidy/option_root_dir/src/test.CC:5:3: warning: Assigned value is garbage or undefined [clang-analyzer-core.uninitialized.Assign]
...
```
When I loop over the files and call `capture3!()` with a single file, I get the same result.

(2) Only two issues reported from `clang_tidy.rb`.
"/tmp/d20200710-6-zwq123/src/foo/テスト.C++:6:5: warning: 2nd function call argument is an uninitialized value [clang-analyzer-core.CallAndMessage]\n
...
/tmp/d20200710-6-zwq123/src/test.CC:5:3: warning: Assigned value is garbage or undefined [clang-analyzer-core.uninitialized.Assign]\n
..."
With literal arguments `stdout, = capture3!(analyzer_bin, "baz/test.cp", "foo/テスト.C++", "test.CC", "--", *option_include_path,`, I got the same results.

I cannot find the cause of this problem so far. I will give up your suggestion.

Reply:
OK. My suggestion is nice to have. Not must. 👍